data/core/common.lua

```diff
@@ -32,18 +61,44 @@ function common.merge(a, b)
 end
 
 
+---
+---Returns the value of a number rounded to the nearest integer.
+---
+---@param n number
+---
+---@return number rounded_number
 function common.round(n)
   return n >= 0 and math.floor(n + 0.5) or math.ceil(n - 0.5)
 end
 
 
+---
+---Returns the first index where a subtable in tbl has prop set.
+---If none is found, nil is returned.
+---
+---@param tbl table
+---@param prop any
+---
+---@return number|nil index
 function common.find_index(tbl, prop)
   for i, o in ipairs(tbl) do
     if o[prop] then return i end
   end
 end
 
 
+---
+---Returns a value between a and b on a linear scale, based on the
+---interpolation point t.
+---
+---If a and b is a table, a table containing the result for all the
+---elements in a and b is returned.
+---
+---@param a number|table
+---@param b number|table
+---@param t number
+---
+---@return number|table interpolated_value
```

Comment:
It might be better to either use `@overload` or to add another function annotation to clarify the usage with tables.